### PR TITLE
Introduce connectAndSignIn

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: packages/wallet-sdk/src/relay/Web3Request.ts [hunk 1]
```diff
@@ -21,9 +21,26 @@ interface BaseWeb3Request<
...
+  RequestEthereumAccountsBaseParams & {
+    domain: string;
+    aud: string;
+    version: string;
+    type: string;
+    nonce: string;
+    iat: string;
+    chainId: string;
+    statement?: string;
+    resources?: string[];
```
Comment (bangtoven) [inline packages/wallet-sdk/src/relay/Web3Request.ts:43]: it includes `RequestEthereumAccountsBaseParams`

File: packages/wallet-sdk/src/relay/Web3Response.ts [hunk 2]
```diff
@@ -26,6 +26,12 @@ export type RequestEthereumAccountsResponse = BaseWeb3Response<
   AddressString[] // an array of ethereum addresses
 >;
 
+export type ConnectAndSignInResponse = BaseWeb3Response<{
+  addresses: AddressString[];
+  message: string;
+  signature: HexString;
+}>;
+
```
Comment (bangtoven) [inline packages/wallet-sdk/src/relay/Web3Response.ts:34]: open to suggestions

File: packages/wallet-sdk/src/relay/WalletLinkRelay.ts [hunk 3]
```diff
@@ -858,6 +860,31 @@ export class WalletLinkRelay extends WalletSDKRelayAbstract {
     }
   }
 
+  // WIP
```
Comment (bangtoven) [inline packages/wallet-sdk/src/relay/WalletLinkRelay.ts:863]: having it private for now.
will be exposed as public once it's done.

File: packages/wallet-sdk/src/relay/WalletLinkRelay.ts [hunk 4]
```diff
@@ -858,6 +860,33 @@ export class WalletLinkRelay extends WalletSDKRelayAbstract {
...
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  private connectAndSignIn(_params: {
```
Comment (vishnumad) [inline packages/wallet-sdk/src/relay/WalletLinkRelay.ts:866]: nit: `_params` to `params`

**Review by vishnumad — APPROVED** [review]

**Pull request merged**